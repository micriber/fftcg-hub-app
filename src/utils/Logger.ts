import * as Sentry from '@sentry/react-native';

class Logger {
  private transporter: Transporter;
  private static instance: Logger;

  private constructor(loggerTransporter: Transporter) {
    this.transporter = loggerTransporter;
  }

  public static createLogger(transporter: Transporter): Logger {
    if (!Logger.instance) {
      Logger.instance = new Logger(transporter);
    }
    return Logger.instance;
  }

  public static getInstance(): Logger {
    return Logger.instance;
  }

  log(message: string) {
    this.transporter.log(message);
  }

  error(error: Error) {
    this.transporter.error(error);
  }
}

interface Transporter {
  log(message: string): void;
  error(error: Error): void;
}

class ConsoleTransporter implements Transporter {
  log(message: string): void {
    console.log(message);
  }

  error(error: Error): void {
    console.error(error);
  }
}

class SentryTransporter implements Transporter {
  log(message: string): void {
    Sentry.captureMessage(message);
  }

  error(error: Error): void {
    Sentry.captureException(error);
  }
}

Logger.createLogger(
  __DEV__ ? new ConsoleTransporter() : new SentryTransporter(),
);
export default Logger.getInstance();
