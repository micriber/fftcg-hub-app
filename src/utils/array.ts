export const range = (start: number, stop: number, step = 1) =>
  Array(Math.ceil((stop - start) / step) + 1)
    .fill(start)
    .map((x, y) => x + y * step);
