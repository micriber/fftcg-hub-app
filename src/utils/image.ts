type lang = 'fr' | 'eg' | 'de';

export const getCardImageUrl = (
  code: string,
  type = 'full',
  lang: lang = 'fr',
) => `https://fftcg.cdn.sewest.net/images/cards/${type}/${code}_${lang}.jpg`;
