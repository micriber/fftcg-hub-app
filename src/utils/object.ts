type GenericObject = {[key: string]: any};

export function deepEqual<T extends GenericObject, U extends GenericObject>(
  object1: T,
  object2: U,
) {
  const keys1 = Object.keys(object1);
  const keys2 = Object.keys(object2);

  if (keys1.length !== keys2.length) {
    return false;
  }

  for (const key of keys1) {
    const val1 = object1[key];
    const val2 = object2[key];
    const areObjects = isObject(val1) && isObject(val2);
    if (
      (areObjects && !deepEqual(val1, val2)) ||
      (!areObjects && val1 !== val2)
    ) {
      return false;
    }
  }

  return true;
}

export function isObject(object: any) {
  return object != null && typeof object === 'object';
}
