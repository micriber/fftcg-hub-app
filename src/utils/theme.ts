import {DarkTheme as RNDarkTheme} from '@react-navigation/native';

export const DarkTheme = {
  ...RNDarkTheme,
  colors: {
    ...RNDarkTheme.colors,
    // primary: 'rgb(63, 197, 107)',
  },
};
