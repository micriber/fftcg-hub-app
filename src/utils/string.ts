import {range} from './array';

export function makeID(length = 6): string {
  const characters =
    'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
  const charactersLength = characters.length;
  const result = range(1, length)
    .map(() => characters.charAt(Math.floor(Math.random() * charactersLength)))
    .join('');
  return result;
}
