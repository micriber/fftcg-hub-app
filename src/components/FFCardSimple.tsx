import React from 'react';
import {
  Dimensions,
  ImageStyle,
  StyleSheet,
  TouchableOpacity,
  View,
  ViewStyle,
} from 'react-native';
import FastImage from 'react-native-fast-image';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {getCardImageUrl} from '../utils/image';
import {Card} from '../services/api/card';
import replaceTextByIconOrStyle from '../utils/icon';
import {Text} from 'react-native-paper';

type Props = {
  displayOwnPin?: boolean;
  card: Card;
  onPress?: (card: Card) => void;
  onLongPress?: () => void;
  imageStyle?: ImageStyle;
  viewType: ViewType;
  containerStyle?: ViewStyle;
};

type ViewType = 'single' | 'list' | 'grid';

const w = Dimensions.get('window');
// orientation must fixed
const SCREEN_WIDTH = w.width < w.height ? w.width : w.height;
const SCREEN_HEIGHT = w.width < w.height ? w.height : w.width;

const FFCardSimple = ({
  card,
  viewType,
  onPress,
  onLongPress,
  displayOwnPin = false,
  imageStyle,
  containerStyle,
}: Props) => {
  const [isFallbackImage, setIsFallbackImage] = React.useState(false);
  const [src, setSrc] = React.useState(
    getCardImageUrl(card.code, 'full', 'fr'),
  );
  const enSrc = getCardImageUrl(card.code, 'full', 'eg');
  const imageSize = {
    width: SCREEN_WIDTH / 2.2,
    height: SCREEN_HEIGHT / 3.3,
  };
  return (
    <View
      testID="SimpleFFCard"
      key={card.code}
      style={[
        viewType === 'list'
          ? styles.cardListContainer
          : styles.cardGridContainer,
        containerStyle,
      ]}>
      <TouchableOpacity
        activeOpacity={0.3}
        onPress={onPress ? () => onPress(card) : undefined}
        onLongPress={onLongPress}>
        {/*<Surface*/}
        {/*  style={[*/}
        {/*    isListView ? styles.cardListContainer : styles.cardGridContainer,*/}
        {/*    styles.surface,*/}
        {/*    containerStyle,*/}
        {/*    imageSize,*/}
        {/*  ]}>*/}
        <FastImage
          style={[
            imageSize,
            // TODO: Find the correct type for this style
            imageStyle as any,
          ]}
          source={{uri: src}}
          resizeMode={viewType === 'list' ? 'stretch' : 'contain'}
          onError={() => {
            if (!isFallbackImage) {
              setSrc(enSrc);
              setIsFallbackImage(true);
            }
          }}
          testID={`${viewType}-Image-${card.code}`}
        />
        {displayOwnPin && card.userCard.length > 0 && (
          <Icon name="check" style={styles.ownPin} size={20} />
        )}
      </TouchableOpacity>
      {viewType === 'list' && (
        <View style={[styles.cardDescription]}>
          <Text>Code: {card.code}</Text>
          <Text>Nom: {card.name}</Text>
          <Text>Type: {card.type}</Text>
          <Text>Element: {replaceTextByIconOrStyle(card.element)}</Text>
          <Text />
          <Text numberOfLines={6} style={styles.cardTextDescription}>
            {replaceTextByIconOrStyle(card.text)}
          </Text>
        </View>
      )}
      {/*</Surface>*/}
    </View>
  );
};

const styles = StyleSheet.create({
  cardGridContainer: {
    justifyContent: 'flex-end',
    borderRadius: 5,
  },
  cardListContainer: {
    // width: '100%',
    flexDirection: 'row',
    // alignContent: 'stretch',
    // justifyContent: 'center',
    height: SCREEN_HEIGHT / 3.3,
    // alignItems: 'flex-end',
    // alignItems: 'flex-start',
  },
  cardDescription: {
    flex: 1,
    // justifyContent: 'flex-start',
    height: SCREEN_HEIGHT / 3.3,
    // width: SCREEN_WIDTH / 2.2,
  },
  cardTextDescription: {flex: 1},
  ownPin: {
    color: '#238F23',
    elevation: 1,
  },
  surface: {
    // alignItems: 'center',
    // justifyContent: 'center',
    // elevation: 4,
  },
});

export default FFCardSimple;
