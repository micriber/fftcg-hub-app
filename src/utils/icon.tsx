import React from 'react';
import {StyleSheet, Text} from 'react-native';
import reactStringReplace from 'react-string-replace';
import {compose} from './fp';
import GameIcon from '../components/icons/GameIcon';
import {ElementIconFile, GameActionIconFile} from '../enums/element';
import {makeID} from './string';

const circleEnum: {[number: string]: string} = {
  '0': '\u2775',
  '1': '\u2776',
  '１': '\u2776',
  '2': '\u2777',
  '3': '\u2778',
  '4': '\u2779',
  '5': '\u277a',
  '6': '\u277b',
  '7': '\u277c',
  '8': '\u277d',
  '9': '\u277e',
};

const styles = StyleSheet.create({
  bravoure: {fontWeight: 'bold'},
});

const _replaceTextByIcon = (
  typeToMatch: string | RegExp,
  iconReplacement: ElementIconFile | GameActionIconFile,
) => (text: string) => {
  return reactStringReplace(text, typeToMatch, () => (
    <GameIcon name={iconReplacement} key={makeID()} />
  ));
};

const _styllishBBCode = (regex: RegExp, style: Object) => (text: string) => {
  return reactStringReplace(text, regex, (match) => {
    if (match === 'br') {
      return ' ';
    } else if (match.includes('.[[br]]')) {
      return '.\n';
    } else if (match.includes('"[[br]]')) {
      return '"\n';
    } else if (match === 'Bravoure[[br]] ') {
      return (
        <Text key={makeID()} style={styles.bravoure}>
          Bravoure{'\n'}
        </Text>
      );
    } else if (match === '&middot;') {
      return '.';
    } else {
      return (
        <Text key={makeID()} style={[style]}>
          {reactStringReplace(match, '[[br]]   ', () => ' ')}
        </Text>
      );
    }
  });
};

const _styllishNumber = (regex: RegExp, style: Object) => (text: string) => {
  return reactStringReplace(text, regex, (match) => {
    const circleUnicode: string | null = circleEnum[match];
    return (
      <Text key={makeID()} style={[style]}>
        {circleUnicode}
      </Text>
    );
  });
};

const _replaceWrapper = (fn: Function) => (text: string) => fn(text);

export const replaceFireType = _replaceTextByIcon(
  /(《火》|火)/gm,
  ElementIconFile.FIRE,
);
export const replaceIceType = _replaceTextByIcon(
  /(《氷》|氷)/gm,
  ElementIconFile.ICE,
);
export const replaceWindType = _replaceTextByIcon(
  /(《風》|風)/gm,
  ElementIconFile.WIND,
);
export const replaceEarthType = _replaceTextByIcon(
  /(《土》|土)/gm,
  ElementIconFile.EARTH,
);
export const replaceLightningType = _replaceTextByIcon(
  /(《雷》|雷)/gm,
  ElementIconFile.LIGHTNING,
);
export const replaceWaterType = _replaceTextByIcon(
  /(《水》|水)/gm,
  ElementIconFile.WATER,
);
export const replaceLightType = _replaceTextByIcon(
  /(《光》|光)/gm,
  ElementIconFile.LIGHT,
);
export const replaceDarkType = _replaceTextByIcon(
  /(《闇》|闇)/gm,
  ElementIconFile.DARK,
);
export const replaceDownArrowAction = _replaceTextByIcon(
  /(《ダル》|ダル)/gm,
  GameActionIconFile.DOWN_ARROW,
);
export const replaceSpecialAction = _replaceTextByIcon(
  '《S》',
  GameActionIconFile.SPECIAL,
);

export const replaceTagS = _styllishBBCode(/\[\[[s]]]([^/]+)\[\[\/]]/gm, {
  color: '#f80',
  fontStyle: 'italic',
  textShadowColor: 'rgba(0, 0, 0, 0.2)',
  textShadowOffset: {width: -1, height: 1},
  textShadowRadius: 10,
});

export const replaceTagEX = _styllishBBCode(/\[\[ex]]([^/]+)\[\[\/]]/gm, {
  color: '#332a9d',
  fontStyle: 'italic',
  fontSize: 15,
  textShadowColor: 'rgba(0, 0, 0, 0.2)',
  textShadowOffset: {width: -1, height: 1},
  textShadowRadius: 10,
});

export const replaceTagI = _styllishBBCode(/\[\[i]]([^/]+)\[\[\/]]/gm, {
  fontStyle: 'italic',
});

export const replaceTagPBR = _styllishBBCode(
  /(\.\[\[br]] {0,3}|"\[\[br]] {0,3}|Bravoure\[\[br]] )/gm,
  {},
);
export const replaceTagBR = _styllishBBCode(/\[\[(br)]] {0,10}/gm, {});

export const replaceDot = _styllishBBCode(/(&middot;)/gm, {});

export const replaceNumber = _styllishNumber(/《([123456789１])》/gm, {});

const replaceTextByIconOrStyle = (text: string) =>
  compose(
    _replaceWrapper(replaceFireType),
    _replaceWrapper(replaceIceType),
    _replaceWrapper(replaceWindType),
    _replaceWrapper(replaceEarthType),
    _replaceWrapper(replaceLightningType),
    _replaceWrapper(replaceWaterType),
    _replaceWrapper(replaceLightType),
    _replaceWrapper(replaceDarkType),
    _replaceWrapper(replaceDownArrowAction),
    _replaceWrapper(replaceSpecialAction),
    _replaceWrapper(replaceTagEX),
    _replaceWrapper(replaceTagS),
    _replaceWrapper(replaceDot),
    _replaceWrapper(replaceTagBR),
    _replaceWrapper(replaceTagPBR),
    _replaceWrapper(replaceNumber),
    _replaceWrapper(replaceTagI),
  )(text);

export default replaceTextByIconOrStyle;
