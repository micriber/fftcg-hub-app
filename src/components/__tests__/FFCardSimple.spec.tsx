/**
 * @format
 */

import 'react-native';
import React from 'react';
import FFCardSimple from '../FFCardSimple';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import {getCardImageUrl} from '../../utils/image';

const card = {
  code: '1-001H',
  name: 'Boulou',
  text: 'billy 《火》《1》《ダル》',
  type: 'blue',
  element: '火',
  id: '1234-1234-1234-1234',
  rarity: 'H',
  cost: '1000',
  power: '',
  job: 'Soutien',
  category1: '',
  category2: '',
  multicard: '',
  exBurst: '',
};

// More about testing in this repository:
// https://github.com/bamlab/react-native-testing/blob/master/src/pages/Home/__tests__/Home.test.tsx
describe('[Component] FFCardSimple', () => {
  it('renders correctly in grid view', async () => {
    const root = render(<FFCardSimple card={card} isListView={false} />);

    expect(root.toJSON()).toMatchSnapshot();
  });

  it('renders correctly in list view', async () => {
    const root = render(<FFCardSimple card={card} isListView={true} />);

    expect(root.toJSON()).toMatchSnapshot();
  });

  it('renders correctly but with fallback image', async () => {
    const Sample = () => <FFCardSimple card={card} isListView={false} />;
    const root = render(<Sample />);
    fireEvent(root.getByTestId(`Image-${card.code}`), 'onError');
    root.update(<Sample />);
    await waitFor(() => root.getByTestId(`Image-${card.code}`));
    const element = root.getByTestId(`Image-${card.code}`);
    console.log(element.props.source.uri);
    expect(element.props.source.uri).toBe(
      getCardImageUrl(card.code, 'full', 'eg'),
    );

    expect(root.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with twice callback', async () => {
    const Sample = () => <FFCardSimple card={card} isListView={false} />;
    const root = render(<Sample />);
    fireEvent(root.getByTestId(`Image-${card.code}`), 'onError');
    root.update(<Sample />);
    await waitFor(() => root.getByTestId(`Image-${card.code}`));
    fireEvent(root.getByTestId(`Image-${card.code}`), 'onError');
    root.update(<Sample />);
    await waitFor(() => root.getByTestId(`Image-${card.code}`));
    const element = root.getByTestId(`Image-${card.code}`);
    console.log(element.props.source.uri);
    expect(element.props.source.uri).toBe(
      getCardImageUrl(card.code, 'full', 'eg'),
    );

    expect(root.toJSON()).toMatchSnapshot();
  });
});
