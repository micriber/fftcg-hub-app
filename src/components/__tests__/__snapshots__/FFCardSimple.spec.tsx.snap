// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] FFCardSimple renders correctly but with fallback image 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
      Object {
        "alignItems": "center",
        "flex": 1,
      },
    ]
  }
  testID="SimpleFFCard"
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "borderRadius": 10,
              "borderWidth": 1,
              "elevation": 5,
              "height": 255,
              "width": 182.14285714285717,
            },
            undefined,
            Object {
              "flex": 1,
            },
          ],
        ]
      }
    >
      <FastImageView
        onFastImageError={[Function]}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://fftcg.cdn.sewest.net/images/cards/full/1-001H_eg.jpg",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="grid-Image-1-001H"
      />
    </View>
  </View>
</View>
`;

exports[`[Component] FFCardSimple renders correctly in grid view 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
      Object {
        "alignItems": "center",
        "flex": 1,
      },
    ]
  }
  testID="SimpleFFCard"
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "borderRadius": 10,
              "borderWidth": 1,
              "elevation": 5,
              "height": 255,
              "width": 182.14285714285717,
            },
            undefined,
            Object {
              "flex": 1,
            },
          ],
        ]
      }
    >
      <FastImageView
        onFastImageError={[Function]}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://fftcg.cdn.sewest.net/images/cards/full/1-001H_fr.jpg",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="single-Image-1-001H"
      />
    </View>
  </View>
</View>
`;

exports[`[Component] FFCardSimple renders correctly in list view 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "marginBottom": 10,
      },
      undefined,
      Object {
        "alignItems": "center",
        "flex": 1,
      },
    ]
  }
  testID="SimpleFFCard"
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "borderRadius": 10,
              "borderWidth": 1,
              "elevation": 5,
              "height": 255,
              "width": 182.14285714285717,
            },
            undefined,
            Object {
              "flex": 1,
            },
          ],
        ]
      }
    >
      <FastImageView
        onFastImageError={[Function]}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://fftcg.cdn.sewest.net/images/cards/full/1-001H_fr.jpg",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="list-Image-1-001H"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "marginLeft": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          undefined,
        ]
      }
    >
      Code: 
      1-001H
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          undefined,
        ]
      }
    >
      Nom: 
      Boulou
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          undefined,
        ]
      }
    >
      Type: 
      blue
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          undefined,
        ]
      }
    >
      Element: 
      
      <Image
        source="test-file-stub"
        style={
          Array [
            Object {
              "height": 20,
              "width": 20,
            },
            Object {},
          ]
        }
      />
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          undefined,
        ]
      }
    />
    <Text
      numberOfLines={6}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      billy 
      <Image
        source="test-file-stub"
        style={
          Array [
            Object {
              "height": 20,
              "width": 20,
            },
            Object {},
          ]
        }
      />
      
      <Text
        style={
          Array [
            Object {},
          ]
        }
      >
        ‚ù∂
      </Text>
      
      <Image
        source="test-file-stub"
        style={
          Array [
            Object {
              "height": 20,
              "width": 20,
            },
            Object {},
          ]
        }
      />
      
    </Text>
  </View>
</View>
`;

exports[`[Component] FFCardSimple renders correctly with twice callback 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
      Object {
        "alignItems": "center",
        "flex": 1,
      },
    ]
  }
  testID="SimpleFFCard"
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "borderRadius": 10,
              "borderWidth": 1,
              "elevation": 5,
              "height": 255,
              "width": 182.14285714285717,
            },
            undefined,
            Object {
              "flex": 1,
            },
          ],
        ]
      }
    >
      <FastImageView
        onFastImageError={[Function]}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://fftcg.cdn.sewest.net/images/cards/full/1-001H_eg.jpg",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="grid-Image-1-001H"
      />
    </View>
  </View>
</View>
`;
